src/Sarif/Writers/BaseLogger.cs

```diff
@@ -15,9 +15,15 @@ public abstract class BaseLogger
         protected BaseLogger(IEnumerable<FailureLevel> failureLevels,
             IEnumerable<ResultKind> resultKinds)
         {
-            _failureLevels = failureLevels?.ToList() ?? new List<FailureLevel>();
+            _failureLevels = failureLevels?.ToList() ?? new List<FailureLevel>(); ;
             _resultKinds = resultKinds?.ToList() ?? new List<ResultKind>();
 
+            if (_failureLevels.Count == 0 && _resultKinds.Count == 0)
```

Comment:
>_failureLevels [](http://example.com/codeflow?start=16&length=14)

This is just a helper, if you provide us entirely empty data, we will provide you sensible defaults.